modules/runner/src/main/java/org/apache/ignite/internal/configuration/storage/DistributedConfigurationStorage.java

```diff
@@ -306,7 +306,12 @@ public boolean onUpdate(@NotNull WatchEvent events) {
 
                     long newChangeId = masterKeyEntry.revision();
 
-                    assert newChangeId > changeId.get();
+                    assert newChangeId >= changeId.get();
+
+                    // skip already processed update
+                    if (changeId.get() == newChangeId) {
```

Comment:
Ass was mentioned above it seems that there's nothing to skip because all watches are resisted on start.